be/src/exprs/vectorized/binary_predicate.cpp

```diff
@@ -72,6 +100,35 @@ class VectorizedBinaryPredicate final : public Predicate {
     }
 };
 
+class ArrayPredicate final : public Predicate {
+public:
+    explicit ArrayPredicate(const TExprNode& node) : Predicate(node), _comparator(node.opcode) {}
+    ~ArrayPredicate() override = default;
+
+    Expr* clone(ObjectPool* pool) const override { return pool->add(new ArrayPredicate(*this)); }
+
+    StatusOr<ColumnPtr> evaluate_checked(ExprContext* context, vectorized::Chunk* ptr) override {
+        ASSIGN_OR_RETURN(auto l, _children[0]->evaluate_checked(context, ptr));
+        ASSIGN_OR_RETURN(auto r, _children[1]->evaluate_checked(context, ptr));
+        auto lhs_arr = std::static_pointer_cast<ArrayColumn>(l);
+        auto rhs_arr = std::static_pointer_cast<ArrayColumn>(r);
+
+        ColumnBuilder<TYPE_BOOLEAN> builder(ptr->num_rows());
+        std::vector<int8_t> cmp_result;
+        lhs_arr->compare_column(*rhs_arr, &cmp_result);
+
+        // Convert the compare result (-1, 0, 1) to the predicate result (true/false)
+        for (int8_t cmp : cmp_result) {
+            bool pred_res = _comparator(cmp);
```

Comment:
do you dissamble this code to ensure that simd instruction sequence can be generated from this for-loop?
another thing is that spliting compare proceduce into two procedures cost less time than one compare procedure?

Reply:
fixed